src/openzaak/components/documenten/api/validators.py

```diff
@@ -66,3 +76,68 @@ class UniekeIdentificatieValidator(_UniekeIdentificatieValidator):
 
     def __init__(self):
         super().__init__("bronorganisatie", "identificatie")
+
+
+class RemoteRelationValidator:
+    message = _(
+        "The canonical remote relation still exists, this relation cannot be deleted."
+    )
+    code = "remote-relation-exists"
+
+    def __init__(self, request=None):
+        self.request = request
+
+    def __call__(self, oio: ObjectInformatieObject):
+        # external object
+        if isinstance(oio.object, ProxyMixin):
+            invalid = self._check_remote(oio)
+        else:
+            invalid = self._check_local(oio)
+
+        if invalid:
+            logger.info(
+                "Relation between %s and informatieobject still exists", oio.object_type
+            )
+            raise serializers.ValidationError(self.message, self.code)
+
+    @staticmethod
+    def _check_local(oio: ObjectInformatieObject) -> bool:
+        method_map = {
+            ObjectTypes.zaak: "does_zaakinformatieobject_exist",
+            ObjectTypes.besluit: "does_besluitinformatieobject_exist",
+        }
+        check_method = getattr(oio, method_map[oio.object_type])
+        return check_method()
+
+    def _check_remote(self, oio: ObjectInformatieObject) -> bool:
+        object_url = oio.object._loose_fk_data["url"]
+
+        default_version = settings.REST_FRAMEWORK["DEFAULT_VERSION"]
+        document_url = build_absolute_url(
+            oio.informatieobject.latest_version.get_absolute_api_url(
+                version=default_version
+            ),
+            request=self.request,
+        )
```

Comment:
@SilviaAmAm this works for local documents (without using CMIS), but I remember the `Canonical` construction being difficult w/r to CMIS.

What I'm not sure is:

- given an OIO instance, using CMIS
- what's the best way to get the fully qualified document URL of the EIO

I need this URL for the query parameters used in the `besluitinformatieobject` or `zaakinformatieobject` list call.